docusaurus/docs/iOS/swiftui/audio-recording.md

```diff
@@ -0,0 +1,94 @@
+---
+title: Voice Recording
+---
+
+Stream Chat's SwiftUI SDK allows you to record and share async voice messages in your channels. The voice recordings have a built-in attachment type (as defined [here](https://getstream.io/chat/docs/sdk/ios/uikit/guides/working-with-attachments/)).
+
+:::note
+Voice Recordings on SwiftUI are available since version 4.46.0.
+:::
+
+Voice recording is disabled by default. In order to enable it, you should set the `isVoiceRecordingEnabled` property to `true`, when setting up the `StreamChat` client:
+
+```swift
+let utils = Utils(
+    composerConfig: ComposerConfig(isVoiceRecordingEnabled: true)
+)
+let streamChat = StreamChat(chatClient: chatClient, utils: utils)
+```
+
+## Recording UI Flows
+
+The voice recording feature supports several different UI flows.
+
+### Recording
+
+When the user long presses on the voice recording button longer than 1 second, the recording is started. In that case, while the button is still pressed, the recording view is shown.
+
+The recording view provides the following actions:
+- add the recording to the composer input (invoked when releasing the long press button)
+- slide to cancel (invoked when you drag to the slide to cancel indicator)
+- lock the recording (invoked when you drag towards the lock button)
+
+In order to replace this view with your own implementation, you can implement the following `ViewFactory` method.
+
+```swift
+public func makeComposerRecordingView(
+        viewModel: MessageComposerViewModel,
+        gestureLocation: CGPoint
+) -> some View {
+    CustomRecordingView(viewModel: MessageComposerViewModel, location: gestureLocation)
+```
+
+### Locked View
+
+When the user decides to lock the recording, `LockedView` is presented. It provides the user the following actions:
+- discard the recording - you are back to the initial state.
+- stop the recording - you go into the recording preview state.
```

Comment:
Considering the message above, I think `the user` makes more sense, but not a big deal
```suggestion
- discard the recording - the user is back to the initial state.
- stop the recording - the user goes into the recording preview state.
```